framework/src/markers/BoundaryPreservedMarker.C

```diff
@@ -0,0 +1,64 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BoundaryPreservedMarker.h"
+#include "MooseMeshUtils.h"
+
+#include "libmesh/error_vector.h"
+
+registerMooseObject("MooseApp", BoundaryPreservedMarker);
+
+InputParameters
+BoundaryPreservedMarker::validParams()
+{
+  InputParameters params = Marker::validParams();
+  params.addRequiredParam<BoundaryName>(
+      "preserved_boundary",
+      "The name of the boundary to be preserved. Will try to preserve the boundary during AMR");
+  params.addRequiredParam<MarkerName>(
+      "marker", "The marker name to decide whether to carsen or refine elements.");
+  params.addClassDescription("Marks elements for refinement or coarsening based on the provided "
+                             "marker value, while preserving the given boundary.");
+  return params;
+}
+
+BoundaryPreservedMarker::BoundaryPreservedMarker(const InputParameters & parameters)
+  : Marker(parameters),
+    _marker_name(parameters.get<MarkerName>("marker")),
+    _marker(&getMarkerValue(_marker_name))
+{
+  _mesh.errorIfDistributedMesh("BoundaryPreservedMarker");
```

Comment:
```suggestion
  _mesh.errorIfDistributedMesh(type());
```
This is more general as it allows for the possibility that this class gets sub-typed

Reply:
Cool, thanks!